cpp/modmesh/view/RManager.cpp

```diff
@@ -281,6 +202,187 @@ void RManager::setUpMenu()
             []() {}));
 }
 
+void RManager::setUpCameraControllersMenuItems() const
+{
+    auto * use_orbit_camera = new RAction(
+        QString("Use Orbit Camera Controller"),
+        QString("Use Oribt Camera Controller"),
+        [this]()
+        {
+            qDebug() << "Use Orbit Camera Controller (menu demo)";
+            for (auto subwin : m_mdiArea->subWindowList())
+            {
+                auto * viewer = dynamic_cast<R3DWidget *>(subwin->widget());
+
+                if (viewer == nullptr)
+                    continue;
+
+                viewer->scene()->setOrbitCameraController();
+                viewer->scene()->controller()->setCamera(viewer->camera());
+            }
+        });
+
+    auto * use_fps_camera = new RAction(
+        QString("Use First Person Camera Controller"),
+        QString("Use First Person Camera Controller"),
+        [this]()
+        {
+            qDebug() << "Use First Person Camera Controller (menu demo)";
+            for (auto subwin : m_mdiArea->subWindowList())
+            {
+                auto * viewer = dynamic_cast<R3DWidget *>(subwin->widget());
+
+                if (viewer == nullptr)
+                    continue;
+
+                viewer->scene()->setFirstPersonCameraController();
+                viewer->scene()->controller()->setCamera(viewer->camera());
+            }
+        });
+
+    auto * cameraGroup = new QActionGroup(m_mainWindow);
+    cameraGroup->addAction(use_orbit_camera);
+    cameraGroup->addAction(use_fps_camera);
+
+    use_orbit_camera->setCheckable(true);
+    use_fps_camera->setCheckable(true);
+    use_orbit_camera->setChecked(true);
+
+    m_viewMenu->addAction(use_orbit_camera);
+    m_viewMenu->addAction(use_fps_camera);
+}
+
+void RManager::setUpCameraMovementMenuItems() const
+{
+    auto * reset_camera = new RAction(
+        QString("Reset (esc)"),
+        QString("Reset (esc)"),
+        [this]()
+        {
+            const auto * subwin = m_mdiArea->currentSubWindow();
+            if (subwin == nullptr)
+                return;
+
+            auto * viewer = dynamic_cast<R3DWidget *>(subwin->widget());
+            if (viewer == nullptr || viewer->camera() == nullptr)
+                return;
+
+            viewer->resetCamera();
+        });
+
+    auto * move_camera_up = new RAction(
+        QString("Move camera up (W/⬆)"),
+        QString("Move camera up (W/⬆)"),
+        createCameraMovementItemHandler([](CameraInputState & input)
+                                        { input.tyAxisValue = 1.0; }));
+
+    auto * move_camera_down = new RAction(
+        QString("Move camera down (S/⬇)"),
+        QString("Move camera down (S/⬇)"),
+        createCameraMovementItemHandler([](CameraInputState & input)
+                                        { input.tyAxisValue = -1.0; }));
+
+    auto * move_camera_right = new RAction(
```

Comment:
The order of the movement is wsda.  It is fine, and I like it to align with the Cartesian coordinate direction (w: +y, s:-y, d: +x, a: -x).

But the convention in literature of the directional keys are wsad or wasd.  This is a little bit unintuitive.  Googling "wsda" does not return anything related to the direction keys.

Having said that, I do not want to change the order of wsda.  This note is only for our information.